src/main/java/io/druid/data/input/impl/DimensionsSpec.java

```diff
@@ -18,35 +18,61 @@
 package io.druid.data.input.impl;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Function;
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Iterables;
 import com.google.common.collect.Lists;
-import com.google.common.collect.Ordering;
 import com.google.common.collect.Sets;
 import com.metamx.common.parsers.ParserUtils;
 
-import java.util.Collections;
+import javax.annotation.Nullable;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 
+
 public class DimensionsSpec
 {
-  private final List<String> dimensions;
+  private final List<DimensionSchema> dimensions;
   private final Set<String> dimensionExclusions;
-  private final List<SpatialDimensionSchema> spatialDimensions;
+  private final Map<String, DimensionSchema> dimensionSchemaMap;
+
+  @Deprecated
+  private List<SpatialDimensionSchema> spatialDimensions;
```

Comment:
not sure why we need to retain this global variable, eventually it is set to null. i see we use it in verify() but it should be possible to adjust that.